clash-prelude/src/Clash/Signal/Internal.hs

```diff
@@ -439,6 +451,33 @@ type HasDefinedInitialValues (dom :: Domain) =
 type DomainResetPolarity (dom :: Domain) =
   DomainConfigurationResetPolarity (KnownConf dom)
 
+-- * Time rep
+-- | Gets time in 'Picoseconds' from time in 'Seconds'.
+type Seconds      (s  :: Nat) = Milliseconds (1000 * s)
+-- | Gets time in 'Picoseconds' from time in 'Milliseconds'.
+type Milliseconds (ms :: Nat) = Microseconds (1000 * ms)
+-- | Gets time in 'Picoseconds' from time in 'Microseconds'.
+type Microseconds (us :: Nat) = Nanoseconds  (1000 * us)
+-- | Gets time in 'Picoseconds' from time in 'Nanoseconds'.
+type Nanoseconds  (ns :: Nat) = Picoseconds  (1000 * ns)
+-- | Gets time in 'Picoseconds' from time in 'Picoseconds', essentially 'id'.
+type Picoseconds  (ps :: Nat) = ps
+
+-- | Converts a frequency in Hz to a period in picoseconds. This might lead to rounding
```

Comment:
```suggestion
-- | Converts a frequency in hertz to a period in picoseconds. This might lead to rounding
```

It just looks better if you write out picoseconds...